cupy/sparse/__init__.py

```diff
@@ -16,47 +16,7 @@ def __getattr__(name):
             msg = 'cupy.sparse is deprecated. Use cupyx.scipy.sparse instead.'
             warnings.warn(msg, DeprecationWarning)
             return getattr(cupyx.scipy.sparse, name)
+        raise AttributeError(
+            "module 'cupyx.scipy.sparse' has no attribute '{}'".format(name))
```

Comment:
It could be done by
```python
try:
    attr = getattr(cupyx.scipy.sparse, name)
except AttributeError:
    raise
warnings.warn(msg, DeprecationWarning)
return attr
```

I suppose the intention is below for the better compatibility
```suggestion
            "module 'cupy.sparse' has no attribute '{}'".format(name))
```

Reply:
Hmm, I think we can't do both. If we just `raise` as your suggested, the error message says `module 'cupyx.scipy.sparse' has no attribute 'xyz'`

Reply:
And how about using `!r` format?

Combining my three suggestions,
```python
try:
    attr = getattr(cupyx.scipy.sparse, name)
except AttributeError:
    raise AttributeError("module 'cupy.sparse' has no attribute {!r}".format(name))
warnings.warn(msg, DeprecationWarning)
return attr
```

Reply:
Ah, ok your point was looking up attribute twice is costly.
I'll fix in that way.

Reply:
```py
    def __getattr__(name):
        try:
            attr = getattr(cupyx.scipy.sparse, name)
        except AttributeError as e:
            raise AttributeError(
                "module 'cupy.sparse' has no attribute {!r}".format(
                    name)) from e
        warnings.warn(
            'cupy.sparse is deprecated. Use cupyx.scipy.sparse instead.',
            DeprecationWarning)
        return attr
```

Result:

```
>>> cupy.sparse.foo
Traceback (most recent call last):
  File "/home/kenichi/Development/cupy/cupy/sparse/__init__.py", line 16, in __getattr__
    attr = getattr(cupyx.scipy.sparse, name)
AttributeError: module 'cupyx.scipy.sparse' has no attribute 'foo'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kenichi/Development/cupy/cupy/sparse/__init__.py", line 20, in __getattr__
    name)) from e
AttributeError: module 'cupy.sparse' has no attribute 'foo'
```